frontend/src/components/UserPermissionHistoryTable/index.vue

```diff
@@ -0,0 +1,164 @@
+<script setup lang="ts">
+import { watch } from 'vue';
+import { isAxiosError } from 'axios';
+import { useQuery } from '@tanstack/vue-query';
+import { AppActlApiService } from '@/services/ApiServiceFactory';
+import DataTable from 'primevue/datatable';
+import Column from 'primevue/column';
+
+import { hashRouter } from '@/router';
+import TableSkeleton from '@/components/TableSkeleton';
+import DateCol from '@/components/UserPermissionHistoryTable/DateCol.vue';
+import PermissionDetailsCol from '@/components/UserPermissionHistoryTable/PermissionDetailsCol.vue';
+import ChangePerformerCol from '@/components/UserPermissionHistoryTable/ChangePerformerCol.vue';
+
+
+const props = defineProps<{
+    userId: string;
+    applicationId: string
+}>();
+
+const auditHistoryQuery = useQuery(
+    {
+        queryKey: ['permission-audit-history', { user_id: props.userId, application_id: props.applicationId }],
+        queryFn:
+            () => AppActlApiService.permissionAuditApi
+                .getPermissionAuditHistoryByUserAndApplication(
+                    Number(props.userId), Number(props.applicationId)
+                ).then((res) => res.data),
+        enabled: !!props.userId && !!props.applicationId,
+        staleTime: 0,
+        gcTime: 0,
+    }
+);
+
+// Navigate back if user has no permission to view data.
+watch(() => auditHistoryQuery.error.value, (error) => {
+    if (isAxiosError(error) && error.response?.status === 403) {
+        hashRouter.push('/');
+    }
+});
```

Comment:
For the first comment, while using Vue Router’s Navigation Guards to validate data before routing can work, it often makes the app feel clunky and less responsive. Handling these checks centrally adds unnecessary complexity to the application, similar to what we’ve experienced with our breadcrumbs. The current approach aims to be more user-friendly by handling incorrect or incomplete URL parameters right at the component level, ensuring smoother navigation and a better user experience.

For instance, if you put heavy logic or multiple asynchronous calls in a guard, users will experience a longer delay before the route transition completes, which makes the application feel less responsive. In such cases, users may see nothing happening while the guard is waiting for a response, which contributes to a clunky experience.

Regarding the watch on the query error, this is currently the simplest solution since we don’t have an error page to show users. Redirecting back is a temporary measure to handle users who lack permission, and once we introduce proper error handling pages, we can refine this approach.